### PR TITLE
updated to WinAppSDK 1.3.230502000

### PR DESCRIPTION
**Pull request by pratikone**
This PR does 3 things : 
1. Update WinUICS project template to 1.3.230502000.
2. Make tests pass by modifying code for StyleCop
3. Adds code which support dynamic changing of titlebar theme whenever system theme changes even when the app is running

**Applies to the following platforms:**

- [X] WinUI
- [ ] WPF
- [ ] UWP

### CODE REVIEW

File: code/TemplateStudioForWinUICs/Templates/Proj/Default/Param_ProjectName/MainWindow.xaml.cs [hunk 1]
```diff
@@ -4,8 +4,10 @@ namespace Param_RootNamespace;
 
 public sealed partial class MainWindow : WindowEx
 {
-    public Microsoft.UI.Dispatching.DispatcherQueue dispatcherQueue;
-    private UISettings _settings;
+    private Microsoft.UI.Dispatching.DispatcherQueue dQueue;
```
Comment (bpulliam) [inline code/TemplateStudioForWinUICs/Templates/Proj/Default/Param_ProjectName/MainWindow.xaml.cs:7]: can this name change be a breaking change if the old name was used already?

Comment (pratikone) [reply code/TemplateStudioForWinUICs/Templates/Proj/Default/Param_ProjectName/MainWindow.xaml.cs:7]: oh, it is not a breaking change. I committed last change recently.

File: code/test/SharedFunctionality.Tests/BaseGenAndBuildFixture.cs [hunk 2]
```diff
@@ -423,7 +423,7 @@ public static string GetVsInstallRoot()
             var VsEditions = new List<string> { "Enterprise", "Preview", "Professional", "Community" };
 
             // Try both of these to allow for wonderful inconsistencies in resolving
-            var progFileLocations = new List<string> { Environment.GetEnvironmentVariable("ProgramW6432"), Environment.GetEnvironmentVariable("ProgramFiles") };
+            var progFileLocations = new List<string> { Environment.GetEnvironmentVariable("ProgramW6432"), Environment.GetEnvironmentVariable("ProgramFiles"), "D:\\Program Files" };
```
[the remark this answers is not in the document]
Comment (pratikone) [reply code/test/SharedFunctionality.Tests/BaseGenAndBuildFixture.cs:426]: It is for finding VS's installed location. Helps run the test locally easy.  A good solution will be to use `vswhere` but this is just smaller easier solution. Can explore vswhere in the future.

[the remark this answers is not in the document]
Comment (pratikone) [reply code/test/SharedFunctionality.Tests/BaseGenAndBuildFixture.cs:427]: added a new issue and linked it here in the comments. i am keeping the current change  for running tests locally.

File: code/TemplateStudioForWinUICs/Templates/Proj/Default/Param_ProjectName/MainWindow.xaml.cs [hunk 3]
```diff
@@ -15,17 +17,17 @@ public MainWindow()
         Title = "AppDisplayName".GetLocalized();
 
         // Theme change code picked from https://github.com/microsoft/WinUI-Gallery/pull/1239
-        dispatcherQueue = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
-        _settings = new UISettings();
-        _settings.ColorValuesChanged += _settings_ColorValuesChanged; // cannot use FrameworkElement.ActualThemeChanged event because the triggerTitleBarRepaint workaround no longer works
+        dQueue = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
```
Comment (mikebattista) [inline code/TemplateStudioForWinUICs/Templates/Proj/Default/Param_ProjectName/MainWindow.xaml.cs:20]: `dispatcherQueue`

File: code/test/TemplateStudioForWinUICs.Tests/TestData/WinUI/SC/StyleCop/Param_ProjectName/GlobalSuppressions.cs [hunk 4]
```diff
@@ -23,3 +23,4 @@
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1602:Enumeration items should be documented", Justification = "This is a newly added analyzer check--suppressing as so much existing template code fails this rule.")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1633:File must have header", Justification = "By default we do not include a header in the geenrated files.")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1652:Enable XML documentation output", Justification = "We leave this as a choice for the developer to enable. As the generated code doesn't include XML comments, enabling this would just create more issues with the generated code.")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.MaintainabilityRules", "SA1401:FieldsMustBePrivate", Justification = "Sometimes, it makes sense to have public member variables.")]
```
Comment (mikebattista) [inline code/test/TemplateStudioForWinUICs.Tests/TestData/WinUI/SC/StyleCop/Param_ProjectName/GlobalSuppressions.cs:26]: Revert this. Public members should be properties.

Comment (pratikone) [reply code/test/TemplateStudioForWinUICs.Tests/TestData/WinUI/SC/StyleCop/Param_ProjectName/GlobalSuppressions.cs:26]: it is a temporary change. but change to property is simple enough. i can do that

**Review by bpulliam — APPROVED** [review]

**Pull request merged**